src/index.js

```diff
@@ -272,7 +278,12 @@ export default class Carousel extends Component {
    * @param {String} direction - The direction to transition, should be 'right' or 'left'.
    */
   goToSlide(index, direction) {
-    const { beforeChange, transitionDuration, transition } = this.props;
+    const { beforeChange, transitionDuration, transition, onControlClick } = this.props;
+
+    if (onControlClick) {
+      onControlClick();
```

Comment:
Is this the right place for this? Can you add some unit tests showing that the auto-scroll of the carousel doesn't trigger this? From my understanding you wanted the ability to know when the user clicked vs auto-scroll. So maybe it could still live here, but a reason passed along?

Reply:
Good catch, fixing it.